src/utils/auth_utils.js

```diff
@@ -42,11 +42,113 @@ const NPOBackend = axios.create({
   withCredentials: true,
 });
 
+// Sets a cookie in the browser
+const setCookie = (key, value, config) => {
+  let cookie = `${key}=${value}; max-age=${config.maxAge}; path=${config.path}`;
+
+  if (config.domain) {
+    cookie += `; domain=${config.domain}`;
+  }
+  if (cookie.secure) {
+    cookie += '; secure';
+  }
+  document.cookie = cookie;
+};
+
+/**
+ * Returns the current user synchronously
+ * @param {Auth} authInstance
+ * @returns The current user (or undefined)
+ */
+const getCurrentUser = authInstance =>
+  new Promise((resolve, reject) => {
+    const unsubscribe = authInstance.onAuthStateChanged(user => {
+      unsubscribe();
+      resolve(user);
+    }, reject);
+  });
+
 const addRoleToCookies = async cookies => {
   const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
   cookies.set(cookieKeys.ROLE, user.data.user.role, cookieConfig);
 };
 
+// Refreshes the current user's access token by making a request to Firebase
+const refreshToken = async () => {
+  const currentUser = await getCurrentUser(auth);
+  if (currentUser) {
+    const refreshUrl = `https://securetoken.googleapis.com/v1/token?key=${process.env.REACT_APP_FIREBASE_APIKEY}`;
+    const refreshT = currentUser.refreshToken;
+    try {
+      const {
+        data: { access_token: idToken },
+      } = await axios.post(refreshUrl, {
+        grant_type: 'refresh_token',
+        refresh_token: refreshT,
+      });
+      // Sets the appropriate cookies after refreshing access token
+      setCookie(cookieKeys.ACCESS_TOKEN, idToken, cookieConfig);
+      const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
+      setCookie(cookieKeys.ROLE, user.data.user.role, cookieConfig);
+      return idToken;
+    } catch (e) {
```

Comment:
no need for the try/catch since we are catching all errors from the `useEffect` in `ProtectedRoute`